modules/programs/waybar.nix

```diff
@@ -0,0 +1,378 @@
+                # to have them all under one directory structure in the nix store
+                exec = "''${pkgs.writers.writeBashBin "hello-from-waybar" '''
+                  echo "from within waybar"
+                '''}/bin/hello-from-waybar";
+              };
+            }
+          '';
+        };
+
+        margin = lib.mkOption {
+          type = nullOr str;
+          default = null;
+          description = "Margins value using the CSS format without units.";
+          example = "20 5";
+        };
+
+        inherit (margins) margin-top margin-left margin-bottom margin-right;
+
+        name = lib.mkOption {
+          type = nullOr str;
+          default = null;
+          description =
+            "Optional name added as a CSS class, for styling multiple waybars.";
+          example = "waybar-1";
+        };
+
+        gtk-layer-shell = lib.mkOption {
+          type = nullOr bool;
+          default = null;
+          example = false;
+          description =
+            "Option to disable the use of gtk-layer-shell for popups.";
+        };
+      };
+    };
+in {
+  meta.maintainers = [ lib.hm.maintainers.berbiche ];
+
+  options.programs.waybar = with lib.types; {
+    enable = lib.mkEnableOption "Waybar";
+
+    package = lib.mkOption {
+      type = nullOr package;
+      default = pkgs.waybar;
+      defaultText = literalExample "${pkgs.waybar}";
+      description = ''
+        Waybar package to use. Set to <code>null</code> to use the default module.
+      '';
+    };
+
+    settings = lib.mkOption {
+      type = listOf waybarBarConfig;
+      default = [ ];
+      description = ''
+        Configuration for Waybar, see <link
+          xlink:href="https://github.com/Alexays/Waybar/wiki/Configuration"/>
+        for supported values.
+      '';
+      example = literalExample ''
+        [
+          {
+            layer = "top";
+            position = "top";
+            height = 30;
+            # Specify outputs to restrict to certain outputs, otherwise show on all outputs
+            output = [
+              "eDP-1"
+              "DP-1"
+            ];
+            modules-left = [ "sway/workspaces" "sway/mode" "wlr/taskbar" ];
+            modules-center = [ "sway/window" "custom/hello-from-waybar" ];
+            modules-right = [ "mpd" "custom/mymodule#with-css-id" "temperature" ];
+            modules = {
+              "sway/workspaces" = {
+                disable-scroll = true;
+                all-outputs = true;
+              };
+              "custom/hello-from-waybar" = {
+                format = "hello {}";
+                max-length = 40;
+                interval = "once";
+                # You may be interested in using symlinkJoin to merge all scripts in one derivation
+                # to have them all under one directory structure in the nix store
+                exec = "''${pkgs.writers.writeBashBin "hello-from-waybar" '''
```

Comment:
Perhaps something simpler, like
```
exec = pkgs.writeShellScript "hello-from-waybar" '''
  echo "from within waybar"
''';
```
would work?

Or I think you could remove the entire `"custom/hello-from-waybar"` thing? I think the example should be as minimal as possible to give an understanding of what a value should look like. If you want to write more extensive documentation then it would be better done as a chapter inside the manual.

Reply:
Thanks, I replaced the messy `writeBashBin` for the simpler `writeShellScript`. I was unaware that `writeShellScript` returned the output path of the binary